diff -urN libdrm-2.4.65-orig/radeon/r600_pci_ids.h libdrm-2.4.65/radeon/r600_pci_ids.h
--- libdrm-2.4.65-orig/radeon/r600_pci_ids.h	2016-01-04 22:17:05.125852639 +0000
+++ libdrm-2.4.65/radeon/r600_pci_ids.h	2016-01-04 22:20:13.855970577 +0000
@@ -450,6 +450,8 @@
 CHIPSET(0x985E, MULLINS_985E, MULLINS)
 CHIPSET(0x985F, MULLINS_985F, MULLINS)
 
+CHIPSET(0x9920, LIVERPOOL_9920, LIVERPOOL)
+
 CHIPSET(0x1304, KAVERI_1304, KAVERI)
 CHIPSET(0x1305, KAVERI_1305, KAVERI)
 CHIPSET(0x1306, KAVERI_1306, KAVERI)
diff -urN libdrm-2.4.65-orig/radeon/radeon_surface.c libdrm-2.4.65/radeon/radeon_surface.c
--- libdrm-2.4.65-orig/radeon/radeon_surface.c	2016-01-04 22:17:05.125852639 +0000
+++ libdrm-2.4.65/radeon/radeon_surface.c	2016-01-04 22:19:11.827274482 +0000
@@ -82,6 +82,7 @@
     CHIP_BONAIRE,
     CHIP_KAVERI,
     CHIP_KABINI,
+    CHIP_LIVERPOOL,
     CHIP_HAWAII,
     CHIP_MULLINS,
     CHIP_LAST,
